Plugins/Flow.Launcher.Plugin.Program/Views/Models/ProgramSource.cs

```diff
@@ -1,5 +1,91 @@
-﻿
+﻿using System;
+using System.IO;
+using Flow.Launcher.Plugin.Program.Programs;
+
 namespace Flow.Launcher.Plugin.Program.Views.Models
 {
-    public class ProgramSource : Settings.ProgramSource { }
+    /// <summary>
+    /// Contains user added folder location contents as well as all user disabled applications
+    /// </summary>
+    /// <remarks>
+    /// <para>Win32 class applications set UniqueIdentifier using their full file path</para>
+    /// <para>UWP class applications set UniqueIdentifier using their Application User Model ID</para>
+    /// <para>Custom user added program sources set UniqueIdentifier using their location</para>
+    /// </remarks>
+    public class ProgramSource
+    {
+        private string name;
+
+        public string Location { get; set; }
+        public string Name { get => name ?? new DirectoryInfo(Location).Name; set => name = value; }
+        public bool Enabled { get; set; } = true;
+        private string uid { get; set; }
+
+        /// <summary>
+        /// Guranteed lowercase.
+        /// </summary>
+        public string UniqueIdentifier { get => uid; set => uid = value.ToLowerInvariant(); }
+
+        public ProgramSource() { }  // only for json deserialization
+
+        /// <summary>
+        /// Add source by location
+        /// </summary>
+        /// <param name="location">location of program source</param>
+        /// <param name="enabled">enabled</param>
+        public ProgramSource(string location, bool enabled=true)
+        {
+            Location = location;
+            Enabled = enabled;
+            UniqueIdentifier = location;
+        }
+
+        public ProgramSource(ProgramSource source)
```

Comment:
is it referencing the same class? Why do we need such a constructor to copy🤔?

Reply:
Yes. We need it when copying instances.

Reply:
So originally it copies a instance when loading program sources from settings and copying them to the display list. It's been removed now. The only usage now is copying disabled sources to the displaylist (another problem though).